signal_op/extract_events.m

```diff
@@ -0,0 +1,174 @@
+function event_times = extract_events(trace,varargin)
+% event_times = extract_events(trace,varargin)
+%
+% Extract event times from the ICA trace of a cell doing 2 iterations
+%
+%   trace : ica trace of a single cell 
+%   An additional inout argument set to 1 will generate plots 
+% 
+%   event_times : [# of events] x 1 array 
+%
+% Hakan Inan (Mar 15)
+%
+
+%%%%
+%some parameters that can be changed within the script (advanced):
+%%%%
+%mad for df/f trace is multiplied with below to get the coarse threshold
+coarse_thresh_multiplier = 9;
+%mad for deconvolved trace is multiplied with below to get the fine threshold
+fine_thresh_multiplier = 15;
+%offset of the impulse for deconvolution
+offset = 0;
+
+% Length before and after the maximum of the characteristic event, h
+num_frames_before_max = 20;
+num_frames_after_max = 100;
+
+
+idx_event_frames = extract_candidate_events(trace,coarse_thresh_multiplier,0);
+out = iterate_event_indices(idx_event_frames);
+idx_event_frames_v2 = out{1};h_1stround = out{2};
+out = iterate_event_indices(idx_event_frames_v2);
+idx_event_frames_v3 = out{1};h_2ndround = out{2};g_2ndround = out{3};
```

Comment:
"Programming" comment: The above set of iterations lends itself naturally to a for loop. You could then easily specify the number of iterations as a parameter.